pkg/cmdutil/userio/prompt.go

```diff
@@ -77,18 +77,21 @@ func isTerminalInteractive(in io.Reader, out io.Writer) bool {
 
 }
 
-func (s *IOStreams) execute(model tea.Model, opts ...tea.ProgramOption) (tea.Model, error) {
+func (s *IOStreams) Execute(model tea.Model, opts ...tea.ProgramOption) (tea.Model, error) {
 	if _, isWizard := model.(wizard.Model); isWizard || s.CurrentHandler == nil {
 		log.Debug().Msgf("IOStreams.execute: starting new session [%T]", model)
+
 		options := append([]tea.ProgramOption{
 			tea.WithInput(s.in),
 			tea.WithOutput(s.outRaw),
 		}, opts...)
 
-		return tea.NewProgram(
+		model, err := tea.NewProgram(
 			model,
 			options...,
 		).Run()
+		s.CurrentHandler = nil
```

Comment:
What is the reason for setting it to `nil` here?

Reply:
once the currenthandler finishes we want new requests to create their own program instead of try to reuse an existing one (which will no longer exist)

Reply:
But how does it help?
This code is execute in two cases:
- CurrentHandler is already nil
- The passed model is wizard. In this case handler will be updated in the caller function